mungegithub/mungers/milestone-maintainer.go

```diff
@@ -0,0 +1,461 @@
+	notificationMatcher := c.MungerNotificationName(milestoneNotifierName, botName)
+	notifications := c.FilterComments(comments, notificationMatcher)
+	return notifications.GetLast(), true
+}
+
+// labelsCompleteState returns the notification state for an issue
+// that has the required labels to remain in the active milestone.
+func labelsCompleteState(issue *githubapi.Issue, notification *c.Notification, kindLabel, priorityLabel string, sigLabels []string) *milestoneNotification {
+	notificationIsCurrent := notification != nil && notification.Arguments == milestoneLabelsComplete
+	if notificationIsCurrent {
+		return nil
+	}
+
+	mention := mungerutil.GetIssueUsers(issue).AllUsers().Mention().Join()
+	message := approvers.GenerateTemplateOrFail(`{{.mention}}
+
+Issue label settings:
+
+- {{range $index, $sigLabel := .sigLabels}}{{if $index}} {{end}}{{$sigLabel}}{{end}}: Issue will be escaled to these SIGs if needed.
+- {{.priorityLabel}}: {{.priorityDescription}}
+- {{.kindLabel}}: {{.kindDescription}}
+
+{{.detail}}
+`, "message", map[string]interface{}{
+		"mention":             mention,
+		"sigLabels":           sigLabels,
+		"priorityLabel":       priorityLabel,
+		"priorityDescription": priorityMap[priorityLabel],
+		"kindLabel":           kindLabel,
+		"kindDescription":     kindMap[kindLabel],
+		"detail":              commentDetail,
+	})
+	if message == nil {
+		return nil
+	}
+
+	return &milestoneNotification{
+		deleteExisting: notification != nil,
+		description:    milestoneLabelsComplete,
+		message:        *message,
+	}
+}
+
+// labelsIncompleteState returns the notification state for an issue
+// lacking the labels required for the active milestone.
+func labelsIncompleteState(issue *githubapi.Issue, notification *c.Notification, labelErrors []string, commentCreatedAt *time.Time, removeIn *time.Duration) *milestoneNotification {
+	isCurrent := (notification != nil && notification.Arguments == milestoneLabelsIncomplete && (commentCreatedAt == nil || time.Since(*commentCreatedAt) < milestoneWarningPeriod))
+	if isCurrent {
+		return nil
+	}
+
+	mention := mungerutil.GetIssueUsers(issue).AllUsers().Mention().Join()
+	var warning string
+	if removeIn != nil {
+		// Round up to the nearest day to ensure that when 6h is
+		// remaining until removal from the milestone, '1' day is
+		// indicated instead of '0'.
+		//
+		// Ceil can be used because removeIn should always be
+		// positive.  A non-blocker past the grace period
+		// (i.e. removeIn < 0) should result in removal from milestone
+		// rather than a warning.
+		days := int(math.Ceil(removeIn.Hours() / 24))
+		warning = fmt.Sprintf("  If the required changes are not made within %s, the issue will be moved out of the %s milestone.", dayPhrase(days), *issue.Milestone.Title)
+	}
+	message := approvers.GenerateTemplateOrFail(`{{.mention}}
+
+**Action required**: This issue requires label changes.{{.warning}}
+
+{{range $index, $labelError := .labelErrors}}
+{{$labelError}}
+{{end}}
```

Comment:
I *think* this will introduce multiple newlines between each label error.  Not sure if that is what you intended.  You can truncate the extra newlines with a `-` in the template - e.g. `{{range $index, $labelError := .labelErrors -}}` will not include the newline immediately following.